paho-mqtt5 test failing with `Error: Unable to create websockets listener on port 9001`

I cannot run the paho-mqtt5 test. It looks like Testcontainers cannot start the container

```
Running org.apache.camel.quarkus.component.paho.mqtt5.it.PahoMqtt5Test
2022-06-20 17:01:06,720 INFO  [org.jbo.threads] (main) JBoss Threads version 3.4.2.Final
2022-06-20 17:01:07,800 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (pool-3-thread-1) TestcontainersConfiguration(userProperties={docker.client.strategy=org.testcontainers.dockerclient.UnixSocketClientProviderStrategy}, classpathProperties={}, environment={PATH=/home/ppalaga/gems/bin:/home/ppalaga/orgs/graalvm/mx:/home/ppalaga/bin/async-profiler-1.6-linux-x64:/home/ppalaga/bin/vscode-installations/code/bin:/home/ppalaga/.yarn/bin:/home/ppalaga/.config/yarn/global/node_modules/.bin:/home/ppalaga/bin:/home/ppalaga/bin/gradle-3.5.x/bin:/home/ppalaga/gems/bin:/home/ppalaga/.sdkman/candidates/mvnd/current/bin:/home/ppalaga/.sdkman/candidates/maven/current/bin:/home/ppalaga/.sdkman/candidates/jbang/current/bin:/home/ppalaga/.sdkman/candidates/jbake/current/bin:/home/ppalaga/.sdkman/candidates/java/current/bin:/home/ppalaga/orgs/graalvm/mx:/home/ppalaga/bin/async-profiler-1.6-linux-x64:/home/ppalaga/bin/vscode-installations/code/bin:/home/ppalaga/.yarn/bin:/home/ppalaga/.config/yarn/global/node_modules/.bin:/home/ppalaga/bin:/home/ppalaga/bin/gradle-3.5.x/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin, IMSETTINGS_MODULE=none, INVOCATION_ID=a5e28c0b01a74dc4873a2d517263007b, XAUTHORITY=/run/user/1000/xauth_yYFUEH, SDKMAN_VERSION=5.15.0, HISTCONTROL=ignoredups, XMODIFIERS=@im=none, XDG_DATA_DIRS=/home/ppalaga/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share, GTK_IM_MODULE=gtk-im-context-simple, DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus, XDG_CURRENT_DESKTOP=KDE, JOURNAL_STREAM=8:47252, QT_AUTO_SCREEN_SCALE_FACTOR=0, MAVEN_HOME=/home/ppalaga/.sdkman/candidates/maven/3.8.4, MAIL=/var/spool/mail/ppalaga, SSH_AGENT_PID=1881, SDKMAN_DIR=/home/ppalaga/.sdkman, GRAALVM_HOME=/home/ppalaga/.sdkman/candidates/java/21.2.0.r11-grl, SESSION_MANAGER=local/unix:@/tmp/.ICE-unix/1979,unix/unix:/tmp/.ICE-unix/1979, LOGNAME=ppalaga, JBANG_EDITOR=code, MANAGERPID=1565, PWD=/home/ppalaga/orgs/cq/camel-quarkus/integration-tests/paho-mqtt5, XCURSOR_THEME=breeze_cursors, LANGUAGE=en_US, KDE_SESSION_VERSION=5, LESSOPEN=||/usr/bin/lesspipe.sh %s, SHELL=/bin/bash, SDKMAN_CANDIDATES_API=https://api.sdkman.io/2, ECLIPSE_WORKSPACE=/home/ppalaga/orgs/cq, OLDPWD=/home/ppalaga/orgs/cq/camel-quarkus/integration-tests/paho-mqtt5, GEM_HOME=/home/ppalaga/gems, KDE_FULL_SESSION=true, SSH_ASKPASS=/usr/bin/ksshaskpass, XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0, DEBUGINFOD_URLS=https://debuginfod.fedoraproject.org/ , FG_AIRCRAFT=/home/ppalaga/FlightGear/Aircraft, IMSETTINGS_INTEGRATE_DESKTOP=yes, MVND_HOME=/home/ppalaga/.sdkman/candidates/mvnd/current, SYSTEMD_EXEC_PID=2052, LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.m4a=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.oga=01;36:*.opus=01;36:*.spx=01;36:*.xspf=01;36:, XDG_SESSION_DESKTOP=KDE, SHLVL=2, FG_SCENERY=/home/ppalaga/FlightGear/Scenery, KDE_APPLICATIONS_AS_SCOPE=1, OXYGEN_DISABLE_INNER_SHADOWS_HACK=1, QT_IM_MODULE=xim, HISTSIZE=100000, SDKMAN_OLD_PWD=/home/ppalaga/orgs/cq/camel-quarkus/integration-tests/paho-mqtt5, JAVA_HOME=/home/data/jvm/adopt-openjdk/jdk-11, BASH_FUNC_which%%=() {  ( alias;
 eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
}, JBANG_HOME=/home/ppalaga/.sdkman/candidates/jbang/current, TERM=xterm, KDE_SESSION_UID=1000, XDG_CONFIG_DIRS=/home/ppalaga/.config/kdedefaults:/etc/xdg:/usr/share/kde-settings/kde-profile/default/xdg, XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0, LANG=en_IE.UTF-8, XDG_SESSION_ID=1, MOZ_GMP_PATH=/usr/lib64/mozilla/plugins/gmp-gmpopenh264/system-installed, XDG_SESSION_TYPE=x11, HISTFILESIZE=100000, DISPLAY=:0, GTK_RC_FILES=/etc/gtk/gtkrc:/home/ppalaga/.gtkrc:/home/ppalaga/.config/gtkrc, which_declare=declare -f, ECLIPSE_HOME=/home/ppalaga/bin/eclipse-installations/java-latest/eclipse, MAVEN_CMD_LINE_ARGS= -DtrimStackTrace=false --no-snapshot-updates -Dmaven.wagon.http.ssl.insecure=true -Dmaven.wagon.http.ssl.allowall=true -Dmaven.wagon.http.ssl.ignore.validity.dates=true clean test, _=/home/data/jvm/adopt-openjdk/jdk-11.0.7+10/bin/java, XDG_SESSION_CLASS=user, SDKMAN_CANDIDATES_DIR=/home/ppalaga/.sdkman/candidates, JBAKE_HOME=/home/ppalaga/.sdkman/candidates/jbake/current, XCURSOR_SIZE=24, LIBOVERLAY_SCROLLBAR=0, GDK_CORE_DEVICE_EVENTS=1, DESKTOP_SESSION=/usr/share/xsessions/plasmax11, MAVEN_PROJECTBASEDIR=/home/ppalaga/orgs/cq/camel-quarkus, SDKMAN_PLATFORM=linuxx64, USER=ppalaga, GTK2_RC_FILES=/etc/gtk-2.0/gtkrc:/home/ppalaga/.gtkrc-2.0:/home/ppalaga/.gtkrc-2.0-kde4:/home/ppalaga/.config/gtkrc-2.0, XDG_MENU_PREFIX=kf5-, XDG_SEAT=seat0, SSH_AUTH_SOCK=/tmp/ssh-XXXXXXaGxYcF/agent.1645, EDITOR=/home/ppalaga/bin/nano-ansi.sh, GTK_OVERLAY_SCROLLING=0, HOSTNAME=terpistone, KDEDIRS=/usr, XDG_RUNTIME_DIR=/run/user/1000, XDG_VTNR=2, HOME=/home/ppalaga})
2022-06-20 17:01:07,820 INFO  [org.tes.uti.ImageNameSubstitutor] (pool-3-thread-1) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2022-06-20 17:01:07,841 INFO  [org.tes.doc.DockerClientProviderStrategy] (pool-3-thread-1) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-06-20 17:01:08,237 INFO  [org.tes.doc.DockerClientProviderStrategy] (pool-3-thread-1) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
2022-06-20 17:01:08,239 INFO  [org.tes.DockerClientFactory] (pool-3-thread-1) Docker host IP address is localhost
2022-06-20 17:01:08,257 INFO  [org.tes.DockerClientFactory] (pool-3-thread-1) Connected to docker: 
  Server Version: 20.10.17
  API Version: 1.41
  Operating System: Fedora Linux 35 (KDE Plasma)
  Total Memory: 64171 MB
2022-06-20 17:01:08,335 INFO  [🐳 .3.3]] (pool-3-thread-1) Creating container for image: testcontainers/ryuk:0.3.3
2022-06-20 17:01:08,469 INFO  [🐳 .3.3]] (pool-3-thread-1) Container testcontainers/ryuk:0.3.3 is starting: ba681acff325f0416ae67cdf9a06a48b028fe0f3e6b3c31bd2eac80eaba8da53
2022-06-20 17:01:08,990 INFO  [🐳 .3.3]] (pool-3-thread-1) Container testcontainers/ryuk:0.3.3 started in PT0.727135S
2022-06-20 17:01:08,997 INFO  [org.tes.uti.RyukResourceReaper] (pool-3-thread-1) Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2022-06-20 17:01:08,997 INFO  [org.tes.DockerClientFactory] (pool-3-thread-1) Checking the system...
2022-06-20 17:01:08,998 INFO  [org.tes.DockerClientFactory] (pool-3-thread-1) ✔︎ Docker server version should be at least 1.6.0
2022-06-20 17:01:08,998 INFO  [🐳 .6.12]] (pool-3-thread-1) Creating container for image: eclipse-mosquitto:1.6.12
2022-06-20 17:01:09,403 INFO  [🐳 .6.12]] (pool-3-thread-1) Container eclipse-mosquitto:1.6.12 is starting: e49cf8c1c9ca6d66d0ca0cae3323139beea5603935114e7a2daf52131bd240c9
2022-06-20 17:01:09,771 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: mosquitto version 1.6.12 starting
2022-06-20 17:01:09,772 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Config loaded from /mosquitto/config/mosquitto.conf.
2022-06-20 17:01:09,772 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Opening ipv4 listen socket on port 1883.
2022-06-20 17:01:09,773 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Opening ipv6 listen socket on port 1883.
2022-06-20 17:01:09,773 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Opening ipv4 listen socket on port 8883.
2022-06-20 17:01:09,773 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Opening ipv6 listen socket on port 8883.
2022-06-20 17:01:09,774 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Opening websockets listen socket on port 9001.
2022-06-20 17:01:09,774 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: libuv support not compiled in
2022-06-20 17:01:09,774 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: OOM allocating 1073741816 fds
2022-06-20 17:01:09,774 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: ZERO RANDOM FD
2022-06-20 17:01:09,774 INFO  [org.apa.cam.qua.com.pah.mqt.it.PahoMqtt5TestResource] (docker-java-stream-593594151) STDERR: 1655737269: Error: Unable to create websockets listener on port 9001.
2022-06-20 17:01:10,100 ERROR [🐳 .6.12]] (pool-3-thread-1) Could not start container: java.lang.IllegalStateException: Container exited with code 1
        at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:509)
        at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:340)
        at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:338)
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:326)
        at org.apache.camel.quarkus.component.paho.mqtt5.it.PahoMqtt5TestResource.start(PahoMqtt5TestResource.java:96)
        at io.quarkus.test.common.TestResourceManager$TestResourceEntryRunnable.run(TestResourceManager.java:452)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)

2022-06-20 17:01:10,127 ERROR [🐳 .6.12]] (pool-3-thread-1) Log output from the failed container:
1655737269: mosquitto version 1.6.12 starting
1655737269: Config loaded from /mosquitto/config/mosquitto.conf.
1655737269: Opening ipv4 listen socket on port 1883.
1655737269: Opening ipv6 listen socket on port 1883.
1655737269: Opening ipv4 listen socket on port 8883.
1655737269: Opening ipv6 listen socket on port 8883.
1655737269: Opening websockets listen socket on port 9001.
1655737269: libuv support not compiled in
1655737269: OOM allocating 1073741816 fds
1655737269: ZERO RANDOM FD
1655737269: Error: Unable to create websockets listener on port 9001.

[ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 4, Time elapsed: 6.471 s <<< FAILURE! - in org.apache.camel.quarkus.component.paho.mqtt5.it.PahoMqtt5Test
```

This blog https://ichbinblau.github.io/2019/09/06/Setup-nofile-for-Docker/ recommends to lower the number of open files for docker.
I wonder if anybody did that on Fedora and how?

--- a/integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java
+++ b/integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java
@@ -19,6 +19,7 @@ package org.apache.camel.quarkus.component.paho.mqtt5.it;
 import java.util.HashMap;
 import java.util.Map;
 
+import com.github.dockerjava.api.model.Ulimit;
 import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
 import org.apache.camel.quarkus.test.AvailablePortFinder;
 import org.apache.camel.util.CollectionHelper;
@@ -89,8 +90,9 @@ public class PahoMqtt5TestResource implements QuarkusTestResourceLifecycleManage
             }
 
             container.withLogConsumer(new Slf4jLogConsumer(LOGGER))
-                    .waitingFor(Wait.forLogMessage(".* mosquitto version .* running", 1))
-                    .waitingFor(Wait.forListeningPort());
+                    .waitingFor(Wait.forLogMessage(".* mosquitto version .* running.*", 1))
+                    .withCreateContainerCmdModifier(
+                            cmd -> cmd.getHostConfig().withUlimits(new Ulimit[] { new Ulimit("nofile", 512L, 512L) }));
 
             if (startContainer) {
                 container.start();

